### PR TITLE
correctly close rows objects if there are multiple result sets

### PR DESCRIPTION
**Pull request by mjibson**

### CODE REVIEW

File: conn_test.go [hunk 1]
```diff
@@ -1583,3 +1583,25 @@ func TestRowsResultTag(t *testing.T) {
...
+	rows, err := db.Query("SELECT 1; SELECT 2;")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := rows.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	var id int
+	if err := db.QueryRow("SELECT 3").Scan(&id); err != nil {
```
Comment (cbandy) [inline conn_test.go]: Should these calls be on a `Tx` to definitely use the same connection?

Comment (mjibson) [inline conn_test.go]: Yes. Done.

Comment (tamird) [inline conn_test.go]: I think you missed this one, which is still a call on `db`.

Comment (mjibson) [inline conn_test.go]: ergh. yes.

**Pull request merged**